frontend/src/SignUp/SignUp.vue

```diff
@@ -0,0 +1,334 @@
+    return {
+      elem: 1,
+      vertical: true,
+      editable: true,
+
+      valid: true,
+      lazy: false,
+      name: '',
+      email: '',
+      password: '',
+      passwordConfirmation: '',
+      photo: null,
+      nameRules: [
+        v => !!v || 'Name is required',
+        v => (v && v.length <= 10) || 'Name must be less than 10 characters'
+      ],
+      emailRules: [
+        v => !!v || 'E-mail is required',
+        v => /.+@.+\..+/.test(v) || 'E-mail must be valid'
+      ],
+      passwordRules: [
+        v => !!v || 'Password is required'
+      ],
+      passwordConfirmationRules: [
+        v => !!v || 'Password confirmation must be equal to password',
+        v => v === this.password || 'Password confirmation must be equal to password'
+      ],
+
+      keywords: [],
+      currentKeyword: '',
+
+      search: '',
+      headers: [
+        {
+          text: 'Name',
+          align: 'left',
+          sortable: false,
+          value: 'name'
+        },
+        { text: 'Symbol', value: 'symbol' },
+        { text: 'Capital Market (€)', value: 'capital' },
+        { text: 'Price (€)', value: 'price' },
+        { text: 'Outstanding Offer', value: 'outstanding' },
+        { text: 'Volume (24h)', value: 'volume' },
+        { text: '% 1h', value: 'hour' },
+        { text: '% 24h', value: 'hour24' },
+        { text: '% 7j', value: 'week' }
+      ],
+      favorites: [
+        {
+          name: 'Bitcoin',
+          symbol: 'BTC',
+          capital: 139115360496,
+          price: 7668.85,
+          outstanding: 18159712,
+          volume: 26456005213,
+          hour: 0.30,
+          hour24: 4.93,
+          week: 8.07
+        },
+        {
+          name: 'Ethereum',
+          symbol: 'ETH',
+          capital: 14754890165,
+          price: 135.03,
+          outstanding: 109271560,
+          volume: 9171025522,
+          hour: 0.75,
+          hour24: 4.49,
+          week: 4.92
+        },
+        {
+          name: 'XRP',
+          symbol: 'XRP',
+          capital: 8647634066,
+          price: 0.199409,
+          outstanding: 43366238611,
+          volume: 1449114597,
+          hour: 2.08,
+          hour24: 4.29,
+          week: 3.14
+        }
+      ],
+      favoritesSelected: [],
+      confirmation: false,
+      formConfirmation: false
+    }
+  },
+  watch: {
+  },
+  methods: {
+    validate1 () {
+      if (this.$refs.form.validate()) {
+        this.elem = 2
+      }
+    },
+    validate2 () {
+      this.elem = 3
+    },
+    submit () {
+      let target = {
+        is_admin: false,
+        username: this.name,
+        email: this.email,
+        password: this.password,
+        picture_url: this.photo,
+        keyword: this.keyword,
+        favorites_crypto: this.favoritesSelected
+      }
+      if (this.$refs.formConfirmation.validate()) {
+        console.log(target)
+        this.elem = 4
+      }
+    },
+    cancel1 () {
+      window.location.reload()
```

Comment:
Little bit ugly... Reset all data might be better since it doesn't load the page again ?

Reply:
ugly ? it rightly reset all data, purpose is reached ?

Reply:
It seems to be better to reset all data to their default value instead of reload the page for many reasons like smooth sailing, performances (all components are mounted in the DOM and reload the page and I'm pretty sure that reset all data is faster than reload the page and mount all the component in the DOM again)...